Chummer/frmCareer.cs

```diff
@@ -16324,6 +16372,261 @@ private void AddCyberwareSuite(Improvement.ImprovementSource objSource)
+            {
+                SustainedSpell objSustained = new SustainedSpell(CharacterObject);
+                objSustained.CreateByMapping(objSpell, objSustained);
+                CharacterObject.SustainedCollection.Add(objSustained);
+            }
+        }
+
+        /// <summary>
+        /// Adds a sustained complex form
+        /// </summary>
+        private void AddSustainedComplexForm()
+        {
+            if (treComplexForms.SelectedNode != null && treComplexForms.SelectedNode.Level > 0 && treComplexForms.SelectedNode.Tag is ComplexForm objComplexForm)
+            {
+                SustainedComplexForm objSustained = new SustainedComplexForm(CharacterObject);
+                objSustained.CreateByMapping(objComplexForm, objSustained);
+                CharacterObject.SustainedCollection.Add(objSustained);
+            }
+        }
+
+        /// <summary>
+        /// Add a sustained critter power
+        /// </summary>
+        private void AddSustainedCritterPower()
+        {
+            if (treCritterPowers.SelectedNode != null && treCritterPowers.SelectedNode.Level > 0 && treCritterPowers.SelectedNode.Tag is CritterPower objCritterPower)
+            {
+                SustainedCritterPower objSustained = new SustainedCritterPower(CharacterObject);
+                objSustained.CreateByMapping(objCritterPower, objSustained);
+                CharacterObject.SustainedCollection.Add(objSustained);
+            }
+        }
+
+        /// <summary>
+        /// Takes in an ISustainable object and gives out the corresponding panel. Returns NULL if No sich panel exists.
+        /// </summary>
+        /// <param name="objSustained"></param>
+        /// <param name="flpSustainedSpells"></param>
+        /// <param name="flpSustainedComplexForms"></param>
+        /// <param name="flpSustainedCritterPowers"></param>
+        /// <returns></returns>
+        private Panel DetermineRefreshingPanel(ISustainable objSustained, Panel flpSustainedSpells, Panel flpSustainedComplexForms, Panel flpSustainedCritterPowers)
+        {
+            Panel refreshingPanel;
+
+            //This C# Version doesn't support cases based on types so this roundabout way is necessary
+            switch (objSustained.GetType().Name)
+            {
+                case nameof(SustainedSpell):
+                    refreshingPanel = flpSustainedSpells;
+                    break;
+
+                case nameof(SustainedComplexForm):
+                    refreshingPanel = flpSustainedComplexForms;
+                    break;
+
+                case nameof(SustainedCritterPower):
+                    refreshingPanel = flpSustainedCritterPowers;
+                    break;
+
+                default:
+                    refreshingPanel = null;
+                    break;
+            }
+
+            return refreshingPanel;
+        }
```

Comment:
This method seems very specific and probably would be cleaner as a local method (i.e. a method that's defined inside another method) instead of even a private one.